python/binding/tritonserver_pybind.cc

```diff
@@ -0,0 +1,1962 @@
+  }
+
+  std::string Id()
+  {
+    const char* val = nullptr;
+    ThrowIfError(TRITONSERVER_InferenceResponseId(triton_object_, &val));
+    return val;
+  }
+
+  uint32_t ParameterCount(uint32_t* count)
+  {
+    uint32_t val = 0;
+    ThrowIfError(
+        TRITONSERVER_InferenceResponseParameterCount(triton_object_, &val));
+    return val;
+  }
+
+  std::tuple<std::string, TRITONSERVER_ParameterType, py::object> Parameter(
+      uint32_t index)
+  {
+    const char* name = nullptr;
+    TRITONSERVER_ParameterType type = TRITONSERVER_PARAMETER_STRING;
+    const void* value = nullptr;
+    ThrowIfError(TRITONSERVER_InferenceResponseParameter(
+        triton_object_, index, &name, &type, &value));
+    py::object py_value;
+    switch (type) {
+      case TRITONSERVER_PARAMETER_STRING:
+        py_value = py::str(reinterpret_cast<const char*>(value));
+        break;
+      case TRITONSERVER_PARAMETER_INT:
+        py_value = py::int_(*reinterpret_cast<const int*>(value));
+        break;
+      case TRITONSERVER_PARAMETER_BOOL:
+        py_value = py::bool_(*reinterpret_cast<const bool*>(value));
+        break;
+      default:
+        throw Unsupported(
+            std::string("Unexpected type '") +
+            TRITONSERVER_ParameterTypeString(type) +
+            "' received as response parameter");
+        break;
+    }
+    return {name, type, py_value};
+  }
+
+  uint32_t OutputCount()
+  {
+    uint32_t val = 0;
+    ThrowIfError(
+        TRITONSERVER_InferenceResponseOutputCount(triton_object_, &val));
+    return val;
+  }
+
+  std::tuple<
+      std::string, TRITONSERVER_DataType, py::array_t<int64_t>, size_t, size_t,
+      TRITONSERVER_MemoryType, int64_t, py::object>
+  Output(uint32_t index)
+  {
+    const char* name = nullptr;
+    TRITONSERVER_DataType datatype = TRITONSERVER_TYPE_INVALID;
+    const int64_t* shape = nullptr;
+    uint64_t dim_count = 0;
+    const void* base = nullptr;
+    size_t byte_size = 0;
+    TRITONSERVER_MemoryType memory_type = TRITONSERVER_MEMORY_CPU;
+    int64_t memory_type_id = 0;
+    void* userp = nullptr;
+    ThrowIfError(TRITONSERVER_InferenceResponseOutput(
+        triton_object_, index, &name, &datatype, &shape, &dim_count, &base,
+        &byte_size, &memory_type, &memory_type_id, &userp));
+    return {name,
+            datatype,
+            py::array_t<int64_t>({dim_count}, {}, shape),
+            reinterpret_cast<size_t>(base),
+            byte_size,
+            memory_type,
+            memory_type_id,
+            reinterpret_cast<PyResponseAllocator::CallbackResource*>(userp)
+                ->user_object};
+  }
+
+  std::string OutputClassificationLabel(uint32_t index, size_t class_index)
+  {
+    const char* val = nullptr;
+    ThrowIfError(TRITONSERVER_InferenceResponseOutputClassificationLabel(
+        triton_object_, index, class_index, &val));
+    return val;
+  }
+};
+
+// forward declaration
+class PyServer;
+
+class PyInferenceRequest
```

Comment:
Is it possible to breakdown this file to one class per file (e.g. `PyInferenceRequest`, `PyServerOptions`, ...)?